src/sentry/services/hybrid_cloud/auth/impl.py

```diff
@@ -130,6 +132,44 @@ def get_organization_key(self, *, key: str) -> Optional[RpcApiKey]:
         except ApiKey.DoesNotExist:
             return None
 
+    def enable_partner_sso(
+        self, *, organization_id: int, provider_key: str, provider_config: Mapping[str, Any]
+    ) -> None:
+        with enforce_constraints(transaction.atomic(router.db_for_write(AuthProvider))):
+            auth_provider = AuthProvider.objects.create(
+                organization_id=organization_id, provider=provider_key, config=provider_config
+            )
+
+            # TODO: Analytics requires a user id
+            # At provisioning time, no user is available so we cannot provide any user
+            # sso_enabled.send_robust(
+            #     organization=sentry_org,
+            #     provider=provider_key,
+            #     sender="EnablePartnerSSO",
+            # )
+
+            AuditLogEntry.objects.create(
+                organization_id=organization_id,
+                actor_label=f"partner_provisioning_api:{provider_key}",
+                target_object=auth_provider.id,
+                event=audit_log.get_event_id("SSO_ENABLE"),
+                data=auth_provider.get_audit_log_data(),
+            )
+
+    def create_auth_identity(
+        self, provider: str, config: Mapping[str, Any], user_id: int, ident: str
+    ) -> None:
+        auth_provider = AuthProvider.objects.get(provider=provider, config=config)
```

Comment:
Do we need to capture the DoesNotExist here? If we don't and this RPC is called from an outbox that message will get wedged.